network/src/commonMain/kotlin/com/wire/kalium/network/api/featureConfigs/FeatureConfigResponse.kt

```diff
@@ -1,91 +1,185 @@
-    DISABLED;
-}
-
-@Serializable
-data class SelfDeletingMessages(
-    @SerialName("config")
-    val config: SelfDeletingMessagesConfig,
-    @SerialName("status")
-    val status: FeatureFlagStatusDTO
+data class MLSConfigDTO(
+    @SerialName("protocolToggleUsers")
+    val protocolToggleUsers: List<String>,
+    @SerialName("defaultProtocol")
+    val defaultProtocol: ConvProtocol,
+    @SerialName("allowedCipherSuites")
+    val allowedCipherSuites: List<Int>,
+    @SerialName("defaultCipherSuite")
+    val defaultCipherSuite: Int
 )
 
 @Serializable
-data class SelfDeletingMessagesConfig(
+data class SelfDeletingMessagesConfigDTO(
     @SerialName("enforcedTimeoutSeconds")
     val enforcedTimeoutSeconds: Int
 )
+
+@JsonClassDiscriminator("name")
+@Serializable
+sealed class FeatureConfigData {
+
+    @Serializable
+    data class Unknown(
+        @SerialName("status")
+        val status: FeatureFlagStatusDTO
+    ) : FeatureConfigData()
+
+    @SerialName("appLock")
+    @Serializable
+    data class AppLock(
+        @SerialName("config")
+        val config: AppLockConfigDTO,
+        @SerialName("status")
+        val status: FeatureFlagStatusDTO
+    ) : FeatureConfigData()
+
+    @SerialName("classifiedDomains")
+    @Serializable
+    data class ClassifiedDomains(
+        @SerialName("config")
+        val config: ClassifiedDomainsConfigDTO,
+        @SerialName("status")
+        val status: FeatureFlagStatusDTO
+    ) : FeatureConfigData()
+
+    @SerialName("conferenceCalling")
+    @Serializable
+    data class ConferenceCalling(
+        @SerialName("status")
+        val status: FeatureFlagStatusDTO
+    ) : FeatureConfigData()
+
+    @SerialName("conversationGuestLinks")
+    @Serializable
+    data class ConversationGuestLinks(
+        @SerialName("status")
+        val status: FeatureFlagStatusDTO
+    ) : FeatureConfigData()
+
+    @SerialName("digitalSignatures")
+    @Serializable
+    data class DigitalSignatures(
+        @SerialName("status")
+        val status: FeatureFlagStatusDTO
+    ) : FeatureConfigData()
+
+    @SerialName("fileSharing")
+    @Serializable
+    data class FileSharing(
+        @SerialName("status")
+        val status: FeatureFlagStatusDTO
+    ) : FeatureConfigData()
+
+    @SerialName("legalhold")
+    @Serializable
+    data class Legalhold(
+        @SerialName("status")
+        val status: FeatureFlagStatusDTO
+    ) : FeatureConfigData()
+
+    @SerialName("searchVisibility")
+    @Serializable
+    data class SearchVisibility(
+        @SerialName("status")
+        val status: FeatureFlagStatusDTO
+    ) : FeatureConfigData()
+
+    @SerialName("selfDeletingMessages")
+    @Serializable
+    data class SelfDeletingMessages(
+        @SerialName("config")
+        val config: SelfDeletingMessagesConfigDTO,
+        @SerialName("status")
+        val status: FeatureFlagStatusDTO
+    ) : FeatureConfigData()
+
+    @SerialName("sndFactorPasswordChallenge")
```

Comment:
Is `snd` a typo?

Reply:
No it's a weird abbreviation used by the backend.